src/checksum.ml

```diff
@@ -60,13 +65,14 @@ let unsafe_digest_32_le ?(off = 0) ~len:top buf =
   done;
   swap16 (lnot !sum land 0xffff)
 
+(* NOTE(dinosaure): only work on 64-bit architecture. *)
 let unsafe_digest_32_be ?(off = 0) ~len:top buf =
   let buf32 = to_int32 ~off ~len:top buf in
   let len = ref top in
   let sum = ref 0 in
   let i = ref 0 in
   while !len >= 4 do
-    let[@warning "-8"] (Some v) = Int32.unsigned_to_int (swap32 buf32.{!i}) in
+    let v = int32_to_int (swap32 buf32.{!i}) in
```

Comment:
May be we should use an `unsafe_*` access. We do the bound check before.

Reply:
ah, in reality, we must use the `.{}` operator to take the advantage of the unboxing by the compiler... I probably should add a comment about that.